flat_mutation_reader.cc

```diff
@@ -912,3 +915,48 @@ class generating_reader final : public flat_mutation_reader::impl {
 flat_mutation_reader make_generating_reader(schema_ptr s, std::function<future<mutation_fragment_opt> ()> get_next_fragment) {
     return make_flat_mutation_reader<generating_reader>(std::move(s), std::move(get_next_fragment));
 }
+
+bool mutation_fragment_stream_validator::operator()(const dht::decorated_key& dk) {
+    // FIXME: Validate partition key monotonicity
+    //        https://github.com/scylladb/scylla/issues/4804
+    return true;
+}
+
+mutation_fragment_stream_validator::mutation_fragment_stream_validator(const schema& s)
+    : _schema(s)
+    , _prev_kind(mutation_fragment::kind::partition_end)
+    , _prev_pos(position_in_partition::end_of_partition_tag_t{})
+{ }
+
+mutation_fragment_stream_validator::~mutation_fragment_stream_validator() {
+    if (_prev_kind != mutation_fragment::kind::partition_end) {
+        on_internal_error(fmr_logger, format("Stream ended with unclosed partition: {}", _prev_kind));
+    }
+}
+
+bool mutation_fragment_stream_validator::operator()(const mutation_fragment& mv) {
+    auto kind = mv.mutation_fragment_kind();
+    auto pos = mv.position();
+    bool valid = false;
+
+    fmr_logger.debug("{}:{}", kind, pos);
```

Comment:
Since there could be many validators active at any time, you should distinguish them. Typically we prefix the line with the value of `this`.

Reply:
ok, will fix in v5.